athena-federation-sdk/src/test/java/com/amazonaws/athena/connector/lambda/data/BlockUtilsPropertiesTest.java

```diff
@@ -0,0 +1,188 @@
+package com.amazonaws.athena.connector.lambda.data;
+
+/*-
+ * #%L
+ * Amazon Athena Query Federation SDK
+ * %%
+ * Copyright (C) 2022 Amazon Web Services
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+
+import com.amazonaws.athena.connector.lambda.data.helpers.ValuesGenerator;
+import com.amazonaws.athena.connector.lambda.data.helpers.FieldsGenerator;
+
+import org.apache.arrow.vector.complex.MapVector;
+import org.apache.arrow.vector.types.Types.MinorType;
+import org.apache.arrow.vector.types.pojo.ArrowType;
+import org.apache.arrow.vector.types.pojo.Field;
+import org.apache.arrow.vector.types.pojo.Schema;
+import org.apache.arrow.vector.util.JsonStringArrayList;
+import org.apache.arrow.vector.util.JsonStringHashMap;
+import org.apache.arrow.vector.FieldVector;
+import org.apache.arrow.memory.RootAllocator;
+import org.apache.arrow.vector.VectorSchemaRoot;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import net.jqwik.api.*;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+
+class BlockUtilsPropertiesTest {
+
+    private static final Logger logger = LoggerFactory.getLogger(BlockUtilsTest.class);
+
+    static final RootAllocator allocator = new RootAllocator();
+
+    @Provide
+    private Arbitrary<Field> fieldLowRecursion() {
+        FieldsGenerator fieldsGenerator = new FieldsGenerator(2);
+        return fieldsGenerator.field();
+    }
+
+    @Provide
+    private Arbitrary<Field> fieldHighRecursion() {
+        FieldsGenerator fieldsGenerator = new FieldsGenerator(5);
+        return fieldsGenerator.field();
+    }
+
+    @Property
```

Comment:
Curious why we are only running this one once, and the other one 2000 times.

Reply:
The default is actually 1000. Since the recursion depth is higher here, it is not possible to cover all the possible combinations so we're using the default number of tries to do a sampling of combinations (whereas the other test with a lower recursion will cover all combinations but cover less depth). I will add a comment for this.